controllers/imagerepository_notifications.go

```diff
@@ -0,0 +1,186 @@
+package controllers
+
+import (
+	"context"
+
+	imagerepositoryv1alpha1 "github.com/konflux-ci/image-controller/api/v1alpha1"
+	"github.com/konflux-ci/image-controller/pkg/quay"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func (r *ImageRepositoryReconciler) AddNotification(notification imagerepositoryv1alpha1.Notifications, imageRepository *imagerepositoryv1alpha1.ImageRepository) (imagerepositoryv1alpha1.NotificationStatus, error) {
```

Comment:
It's still not fixed. If a user input `blah` as method or something, Quay will probably return an error. Then, the error will be returned [here](https://github.com/konflux-ci/image-controller/blob/main/controllers/imagerepository_controller.go#L370) and up to the reconcile. Then retry, endless loop.

Reply:
If Quay API receives a POST request with a notification with a wrong field it returns a 400 error with a descriptive error message. I guess this is considered an "intermittent error" as you named it and f.e. Quay API being down is a "persistent error". That's okay.

I understand your point of the reconcile loop but it is not clear what you want here. Please explain **exactly** how you mean to handle those errors.
You mean f.e. to "ignore" the 400 error from Quay API and **not** return an error so the reconcile loop keeps moving? From what you say, the goal would be to not return an error that could propagate to the reconcile loop. With "proper validation of the user input" you mean that we do perform the validation of the notification structure? Why doing that twice? That's exactly what Quay API does and accordingly returns an error + message.

Plus, I'll add again that this piece of code is a refactor from code that already [exists](https://github.com/konflux-ci/image-controller/blob/main/controllers/imagerepository_controller.go#L260), is currently working as this in image-controller codebase and that was [approved](https://github.com/konflux-ci/image-controller/pull/112).

Reply:
My main point is to avoid endless error loop. Returning or not an error is an implementation detail.
To properly solve the problem we need to distinguish between intermittent (a retry could help) and persistent (retry won't help) errors.

> If Quay API receives a POST request with a notification with a wrong field it returns a 400 error with a descriptive error message. I guess this is considered an "intermittent error" 

It's persistent error, since retry will not change the result/outcome (unlike in case of a network error).

> Plus, I'll add again that this piece of code is a refactor from code that already [exists](https://github.com/konflux-ci/image-controller/blob/main/controllers/imagerepository_controller.go#L260)

We've merged that code because we had tight deadline. It was promised to make it complete and clean.